apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.Tests/Trace/SimpleManagedTracerTest.cs

```diff
@@ -254,6 +256,203 @@ public void MultipleSpans()
+                    IsValidSpan(spans[4], "root");
+            };
+            mockConsumer.Setup(c => c.Receive(Match.Create(rootMatcher)));
+
+            var childThreadsReleased = new ManualResetEventSlim(initialState: false);
+
+            Func<string, string, Task> op = async (childName, grandchildName) =>
+            {
+                childThreadsReleased.Wait();
+
+                using (tracer.StartSpan(childName))
+                {
+                    await Task.Delay(30);
+                    using (tracer.StartSpan(grandchildName))
+                    {
+                        await Task.Delay(30);
+                    }
+                    await Task.Delay(30);
+                }
+            };
+            var t1 = Task.Run(() => op("child-one", "grandchild-one").Wait());
+            var t2 = Task.Run(() => op("child-two", "grandchild-two").Wait());
+
+            using (tracer.StartSpan("root"))
+            {
+                childThreadsReleased.Set();
+                await Task.WhenAll(t1, t2);
+            }
+
+            mockConsumer.VerifyAll();
+        }
+
+        [Fact]
+        public async Task MultipleSpans_Threads_Started_During_Span()
+        {
+            var mockConsumer = new Mock<IConsumer<TraceProto>>();
+            var tracer = SimpleManagedTracer.Create(mockConsumer.Object, ProjectId, TraceId);
+
+            ulong rootId = 0;
+            Predicate<IEnumerable<TraceProto>> rootMatcher = (IEnumerable<TraceProto> t) =>
+            {
+                var spans = t.Single().Spans;
+                rootId = spans[0].SpanId;
+                return spans.Count == 1 && IsValidSpan(spans[0], "root");
+            };
+            mockConsumer.Setup(c => c.Receive(Match.Create(rootMatcher)));
+
+            var childThreadsReleased = new ManualResetEventSlim(initialState: false);
+
+            Func<string, string, Task> op = async (childName, grandchildName) =>
+            {
+                childThreadsReleased.Wait();
+
+                using (tracer.StartSpan(childName))
+                {
+                    await Task.Delay(30);
+                    using (tracer.StartSpan(grandchildName))
+                    {
+                        await Task.Delay(30);
+                    }
+                    await Task.Delay(30);
+                }
+            };
+
+            Task t1;
+            Task t2;
+            using (tracer.StartSpan("root"))
+            {
+                t1 = Task.Run(() => op("child-one", "grandchild-one").Wait());
+                t2 = Task.Run(() => op("child-two", "grandchild-two").Wait());
+            }
+
+            Predicate<IEnumerable<TraceProto>> childMatcher = (IEnumerable<TraceProto> t) =>
+            {
+                // Verify that even though the child spans were started after the root span was ended, they
+                // still have it as a parent since it was opened when their threads were spawned.
```

Comment:
Nit: I'd change this to "when their tasks were created" - there's no guarantee that it'll be starting actual new threads, nor that any new threads would be running by the time we hit the end of the `using` statement. Unless I've misunderstood something, of course - which would probably be related to the previous comment.